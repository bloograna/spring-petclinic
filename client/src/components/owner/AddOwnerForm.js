import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormGroup, FormControl, Col, Form } from 'react-bootstrap';
import CommonForm from '../common/CommonForm';

class AddOwnerForm extends Component {
  // I FUCKING NEED THE REF

  renderGroup = (controlId, type, pattern, label, placeHolder, feedBack) => {
    return (
      <FormGroup as={Col} md="4" controlId={controlId}>
        <Form.Label>{label}</Form.Label>
        <FormControl
          name={controlId}
          required
          type={type ? type : 'text'}
          pattern={pattern ? pattern : '[A-Za-z]+'}
          placeholder={placeHolder}
        />
        <FormControl.Feedback type="invalid">{feedBack}</FormControl.Feedback>
      </FormGroup>
    );
  };

  render() {
    const { formValidated, onChange } = this.props;
    // <Form noValidate validated={formValidated} onSubmit={onSubmit}>
    return (
      <CommonForm onChange={onChange} formValidated={formValidated}>
        <Form.Row>
          {this.renderGroup(
            'firstName',
            null,
            null,
            'First Name',
            'First Name',
            'Please enter a first name, no space allowed'
          )}
          {this.renderGroup(
            'lastName',
            null,
            null,
            'Last Name',
            'Last Name',
            'Please enter a last name, no space allowed'
          )}
        </Form.Row>
        <Form.Row>
          {this.renderGroup(
            'address',
            null,
            '[A-Za-z 0-9]+',
            'Address',
            '',
            'Please enter an address'
          )}
          {this.renderGroup(
            'city',
            null,
            '[A-Za-z ]+',
            'City',
            '',
            'Please enter a city'
          )}
        </Form.Row>
        <Form.Row>
          {this.renderGroup(
            'telephone',
            'tel',
            '[0-9]{10}',
            'Telephone',
            '',
            'Please enter a valid pohne number (10 digits)'
          )}
        </Form.Row>
      </CommonForm>
    );
  }
}

AddOwnerForm.propTypes = {
  formValidated: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired
};

export default AddOwnerForm;
