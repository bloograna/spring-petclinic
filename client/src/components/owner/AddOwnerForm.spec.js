import React from 'react';
import AddOwnerForm from './AddOwnerForm';

describe('AddOwnerForm', () => {
  test('render AddOwnerForm', () => {
    const props = {
      formValidated: true,
      onAddButtonClick: jest.fn(),
      onHideAddOwnerModal: jest.fn()
    };
    const form = shallow(<AddOwnerForm {...props} />);
    expect(form).toMatchSnapshot();
  });
});
