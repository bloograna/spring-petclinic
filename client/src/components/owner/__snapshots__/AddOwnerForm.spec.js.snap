// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddOwnerForm render AddOwnerForm 1`] = `
<CommonForm
  formValidated={true}
  onChange={[Function]}
  onSubmit={[MockFunction]}
>
  <ForwardRef(Bootstrap(FormRow))>
    <ForwardRef(Bootstrap(FormGroup))
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        }
      }
      controlId="firstName"
      md="4"
    >
      <ForwardRef(ContextTransform)>
        First Name
      </ForwardRef(ContextTransform)>
      <ForwardRef(ContextTransform)
        name="firstName"
        pattern="[A-Za-z]+"
        placeholder="First Name"
        required={true}
        type="text"
      />
      <Feedback
        as="div"
        type="invalid"
      >
        Please enter a first name, no space allowed
      </Feedback>
    </ForwardRef(Bootstrap(FormGroup))>
    <ForwardRef(Bootstrap(FormGroup))
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        }
      }
      controlId="lastName"
      md="4"
    >
      <ForwardRef(ContextTransform)>
        Last Name
      </ForwardRef(ContextTransform)>
      <ForwardRef(ContextTransform)
        name="lastName"
        pattern="[A-Za-z]+"
        placeholder="Last Name"
        required={true}
        type="text"
      />
      <Feedback
        as="div"
        type="invalid"
      >
        Please enter a last name, no space allowed
      </Feedback>
    </ForwardRef(Bootstrap(FormGroup))>
  </ForwardRef(Bootstrap(FormRow))>
  <ForwardRef(Bootstrap(FormRow))>
    <ForwardRef(Bootstrap(FormGroup))
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        }
      }
      controlId="address"
      md="4"
    >
      <ForwardRef(ContextTransform)>
        Address
      </ForwardRef(ContextTransform)>
      <ForwardRef(ContextTransform)
        name="address"
        pattern="[A-Za-z 0-9]+"
        placeholder=""
        required={true}
        type="text"
      />
      <Feedback
        as="div"
        type="invalid"
      >
        Please enter an address
      </Feedback>
    </ForwardRef(Bootstrap(FormGroup))>
    <ForwardRef(Bootstrap(FormGroup))
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        }
      }
      controlId="city"
      md="4"
    >
      <ForwardRef(ContextTransform)>
        City
      </ForwardRef(ContextTransform)>
      <ForwardRef(ContextTransform)
        name="city"
        pattern="[A-Za-z ]+"
        placeholder=""
        required={true}
        type="text"
      />
      <Feedback
        as="div"
        type="invalid"
      >
        Please enter a city
      </Feedback>
    </ForwardRef(Bootstrap(FormGroup))>
  </ForwardRef(Bootstrap(FormRow))>
  <ForwardRef(Bootstrap(FormRow))>
    <ForwardRef(Bootstrap(FormGroup))
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        }
      }
      controlId="telephone"
      md="4"
    >
      <ForwardRef(ContextTransform)>
        Telephone
      </ForwardRef(ContextTransform)>
      <ForwardRef(ContextTransform)
        name="telephone"
        pattern="[0-9]{10}"
        placeholder=""
        required={true}
        type="tel"
      />
      <Feedback
        as="div"
        type="invalid"
      >
        Please enter a valid pohne number (10 digits)
      </Feedback>
    </ForwardRef(Bootstrap(FormGroup))>
  </ForwardRef(Bootstrap(FormRow))>
  <ForwardRef(Bootstrap(FormRow))>
    <ForwardRef(Bootstrap(Button))
      onClick={[MockFunction]}
    >
      Cancel
    </ForwardRef(Bootstrap(Button))>
    <ForwardRef(Bootstrap(Button))
      type="submit"
    >
      Add
    </ForwardRef(Bootstrap(Button))>
  </ForwardRef(Bootstrap(FormRow))>
</CommonForm>
`;
